README.md

```diff
@@ -55,87 +47,193 @@ To set up CDF, do the following:
-*Example*: `ts{id=12345678}`
-- You can use various filters and template variables.
-*Example*: `ts{assetIds=[$asset], metadata={key1=~'.*test.*'}, isStep=true}`
-- If you want to apply custom functions to the timeseries (e.g. for unit conversions), simply add your functions around `ts{}`.
-*For example*: `(ts{name=~'Test.*', aggregate='average'} + 10) / 5` 
-    - Click on the info icon next to the text box to see more examples
+But STS query syntax allows us to define aggregation and granularity to each timeseries separately:
+```
+ts{name="test", aggregate="interpolation", granularity="30m"}
+```
+Query above overrides values of aggregation and granularity which is set in UI
+
+**Arithmetic operations**
+
+You're also allowed to apply arithmetic operations for timeseries to combine them. For instance:
+```
+ts{id=123} + ts{externalId="test"}
+```
+Result of the query above is a single plot where datapoints are summed values of each timeseries. Another example:
+```
+ts{name~="test1.*"} + ts{id=123}
+```
+This is another situation, where query `ts{name~="test1.*"}` can return more than 1 timeseries, but let's assume that it returns 3 timeseries with ids `111`, `222` and `333`.
+Then, the result of the query above is 3 plots, which are a combination of summed timeseries values returned by first and second expression in the query. Result plots actually represent queries below:
+- `ts{id=111} + ts{id=123}`
+- `ts{id=222} + ts{id=123}`
+- `ts{id=333} + ts{id=123}`
+
+**Functions**
+STS supports a bunch of functions, that can be applied on timeseries:
+- trigonometric: `sin(ts{})`, `cos(ts{})`, `pi()`
+- variable-length functions: `max(ts{}, ...)`, `min(ts{}, ...)`, `avg(ts{}, ...)`    
+- special: `ln(ts{})`, `pow(ts{}, exponent)`, `sqrt(ts{})`, `exp(ts{})`, `abs(ts{})`   
+- `on_error(ts{}, default)` – for handling errors like overflow or division by zero
+- `map(expression, [list of strings to map from], [list of values to map to], default)` - function, that allows tto work with string timeseries
+
+`on_error()` function allows chart rendering even if some exception appears. It handles errors like:
+- `BAD_DOMAIN` - If bad input ranges are provided. E.g. division by zero, or sqrt of negative number
+- `OVERFLOW` - If result is more than 10^100 in absolute value
+If any of these are encountered, instead of returning a value for that timestamp, CDF returns an error field with an error message.
+To avoid these, you can wrap the (sub)expression in the on_error() function:
+```
+on_error(1/TS{externalId='canBeZero'}, 0)
+```
+
+`map()` function can handle timeseries with string values, i.e. converts strings to doubles. 
+As an example, assume that the time series "stringstate" can have values `"OPEN"` or `"CLOSED"`. Then you can convert it to a number with:
+```
+map(TS{externalId='stringstate'}, ['OPEN', 'CLOSED'], [1, 0], -1)
+```
+`"OPEN"` is mapped to 1, `"CLOSED"` to 0, and everything else to -1
```

Comment:
could you add a link for sts docs and comment with `todo: ` to replace with v1 link when we have it?

Reply:
rn it's in playground

Reply:
`todo` is smthing that needs to be added in code, not in a readme, right?

Reply:
there is no v1 documentation for v1 yet, so i think you could use `<!-- -->` in md files?

Reply:
I know how to add comments to md files, I meant that we probably need to add `todo` in code section, where request to `playground` is done, right?

Reply:
ah, yeah, that we kind of remember. 
but you can add there as well I guess